handlers/secrets.go

```diff
@@ -311,3 +307,27 @@ func removeVolumeMount(volumeName string, mounts []apiv1.VolumeMount) []apiv1.Vo
 func isNotFound(err error) bool {
 	return k8serrors.IsNotFound(err) || k8serrors.IsGone(err)
 }
+
+// processErrorReasons maps k8serrors.ReasonForError to http status codes
+func processErrorReasons(context string, err error) int {
+	switch {
+	case k8serrors.ReasonForError(err) == metav1.StatusReasonConflict:
+		log.Printf("Secret %s error reason: %s, %v\n", context, metav1.StatusReasonConflict, err)
+		return http.StatusConflict
+	case k8serrors.ReasonForError(err) == metav1.StatusReasonInvalid:
+		log.Printf("Secret %s error reason: %s, %v\n", context, metav1.StatusReasonInvalid, err)
+		return http.StatusUnprocessableEntity
+	case k8serrors.ReasonForError(err) == metav1.StatusReasonBadRequest:
+		log.Printf("Secret %s error reason: %s, %v\n", context, metav1.StatusReasonBadRequest, err)
+		return http.StatusBadRequest
+	case k8serrors.ReasonForError(err) == metav1.StatusReasonForbidden:
```

Comment:
I guess my concern with this approach is that we don't give the reason for the error back to the caller. 

We should also be able to test this new method that has been broken out with unit tests when we have a final design on how it is going to work. 

I am not super keen on logging in the function. I'd rather see you return the Status Code plus a message that we then chose to log in the calling function - thereby applying Single Responsibility Principle (SRP)

Reply:
Updated.